src/controllers/albumController.js

```diff
@@ -185,7 +188,31 @@ const getFilteredAlbums = async (req, res) => {
     .skip(parseInt(offset) || 0) // Skip a specified number of albums (pagination implementation)
     .limit(parseInt(limit) || 10); // Limit the number of returned albums (pagination implementation)
 
-  res.status(StatusCodes.OK).json({ albums, count: albums.length }); // Return the found albums and the count of albums
+  const totalPages = Math.ceil(totalCount / (parseInt(limit) || 10));
+  const currentPage = Math.ceil(
+    (parseInt(offset) + 1) / (parseInt(limit) || 10)
+  );
+  const more = currentPage < totalPages;
+
+  res.status(StatusCodes.OK).json({
+    albums,
+    count: albums.length,
+    total: totalCount,
+    more,
+    currentPage,
+    totalPages,
+    nextPage: more
+      ? `${req.originalUrl.split('&offset=')[0]}&offset=${
+          parseInt(offset) + (parseInt(limit) || 10)
+        }`
+      : null,
+    prevPage:
+      currentPage > 1
+        ? `${req.originalUrl.split('&offset=')[0]}&offset=${
+            parseInt(offset) - (parseInt(limit) || 10)
+          }`
+        : null,
```

Comment:
Overall this looks fine but I notice you're splitting on `&offset`. But what if the request was made like this: `/api/v1/albums/filter?offset=5&limit=10&artistName=b` ; I think then your code here won't reconstruct the `nextPage`/\`prevPage\` correctly (though I have not tried it myself yet here)

A preferable way to handle these may be to use the Node built-in (I think at least since version 18 and up) `URL` and `URLSearchParams` modules:

*   [https://nodejs.org/api/url.html#class-urlsearchparams](https://nodejs.org/api/url.html#class-urlsearchparams)
*   [https://www.geeksforgeeks.org/node-js-urlsearchparams-api/](https://www.geeksforgeeks.org/node-js-urlsearchparams-api/)
*   [https://www.youtube.com/watch?v=np53e9_Yilk](https://www.youtube.com/watch?v=np53e9_Yilk)

So we could do something like:

```javascript
const url = new URL(req.originalUrl); // create a "URL" object / class instance from the request
url.searchParams.set('offset': '<whatever the new offset should be>')
// then do url.toString() (or url.href) to get the final updated url string
```

Reply:
Also keep in mind what you want to do if an `offset` was not given here.